.github/workflows/tag_and_release.yml

```diff
@@ -0,0 +1,29 @@
+name: Create tag and release
+
+on:
+  pull_request:
+    paths:
+      - 'CHANGELOG.rst'
+  push:
+    branches:
+      - main
+    paths:
+      - 'CHANGELOG.rst'
+
+jobs:
+  check_version:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check version in changelog
+        run: bash ./tag-from-pipeline.sh verify_changelog_version
```

Comment:
I believe this needs access to the github token? e.g 

```
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Reply:
Also, there is no method of creating the tag.

Reply:
Added github token to the enviroment and method for creating a tag and a release